### DISCUSSION TITLE
Performance Requirements (Google Colab?)

### DISCUSSION THREAD

**Discussion by rosnu** [opendilab/LightZero#269] [General]
Does it make sense to try running LightZero (e.g., minigrid_efficientzero) on Google Colab with a free account? Approximately how long might it take until it's reasonably trained?

**Comment by puyuan1996** [comment]
Indeed, it is possible to run LightZero (e.g., minigrid_muzero) on a free Google Colab account, but several key points need to be considered:

First, the GPU resources provided by Google Colab's free account are typically K80, T4, or P100, which have relatively low performance. Additionally, usage time may be limited, and GPU resources may not always be available. You might need to wait in a queue and could even encounter interruptions during your session.

Based on previous experiments, training minigrid_muzero in the simplest empty8 environment within minigrid takes about 0.2M steps to converge, taking approximately 2 hours on an A100 GPU. Given the lower performance of GPUs on a free Google Colab account, training time could significantly increase. For more complex environments (such as fourroom), convergence on an A100 requires 4M steps, and achieving similar results on less powerful GPUs could take several days. Therefore, if your goal is learning and testing, it is recommended to use the simplest [cartpole environment](https://github.com/opendilab/LightZero/blob/main/zoo/classic_control/cartpole/config/cartpole_muzero_config.py), which typically converges within a few minutes.

If you require faster training speeds and more stable resources, you might need to consider using a more powerful cloud computing platform. Additionally, we are very much looking forward to the latest performance results of minigrid_muzero. If you have any relevant information, please share it. Thank you!

**Reply by rosnu** [reply]
Thank you very much, a very helpful answer, gives me a good impression and starting point. By the way, independent of Colab: Is there a setting for resuming interrupted training in LightZero, or do I need to handle that myself? New versions with date-time extensions are always created, even when I set the model_path.
Thanks
